libs/contrib/Decidable/Decidable/Extra.idr

```diff
@@ -53,21 +53,47 @@ doubleNegationExists {ts} {r} @{dec} nnxs =
   in introduceWitness witness witnessing
 
 
+
+
+decideTransform :
+  {n : Nat}
+  -> {ts : Vect n Type}
+  -> {r : Rel ts}
+  -> {t : Type -> Type}
+  -> (tDec : {a : Type} -> Dec a -> Dec (t a))
+  -> (posDec : Decidable n ts r)
+  -> All ts (chain {ts} Dec (chain {ts} t r))
```

Comment:
Is this return type `Decidable n ts (chain t r)`?

Reply:
Not quite, it's the type of the `decide` field of `Decidable`. My worry was that adding the generic one as an instance would clog up the interface search.

Reply:
Oh I see. That's unfortunate: we should have a name we can use to
talk about a relation being decidable regardless of whether we want
to declare an implementation of the interface.

We could maybe introduce `IsDecidable` and have

```idris
interface Decidable n ts r where
  decide : IsDecidable n ts r
```

?

Reply:
Sounds good. Just pushed with that change